### ISSUE TITLE
Globals in a "with" block

### ISSUE THREAD

**Issue by yguedidi** [twigphp/Twig#2377]
In Symfony, writing:
```twig
{% with {} only %}
    {{ app.request.attributes.get('_route') }}
{% endwith %}
```
leads to a Variable "app" does not exists.

Looks like global Twig variable are not accessible in a `with` block, maybe they should?

**Comment by HeahDude** [comment]
The globals should be available if you remove `only`, aren't they? If so, I don't think this should change.

**Comment by stof** [comment]
@HeahDude yes they are. But this is quite inconsistent with ``{% include ... only %}`` and ``{% embed ... only %}`` where global variables are still available

**Comment by gaiius** [comment]
hi all  
the same with my problem ,
but this my problem ,
please help
http://stackoverflow.com/questions/41886683/how-to-add-extension-add-global-in-twig

**Comment by fabpot** [comment]
fixed in #2933
